model_api/cpp/models/src/anomaly_model.cpp

```diff
@@ -0,0 +1,230 @@
+  imageShape = get_from_any_maps("image_shape", top_priority, mid_priority, imageShape);
+  imageThreshold = get_from_any_maps("image_threshold", top_priority, mid_priority, imageThreshold);
+  pixelThreshold = get_from_any_maps("pixel_threshold", top_priority, mid_priority, pixelThreshold);
+  normalizationScale = get_from_any_maps("normalization_scale", top_priority, mid_priority, normalizationScale);
+  task = get_from_any_maps("task", top_priority, mid_priority, task);
+  labels = get_from_any_maps("labels", top_priority, mid_priority, labels);
+}
+
+AnomalyModel::AnomalyModel(std::shared_ptr<ov::Model> &model,
+                           const ov::AnyMap &configuration)
+    : ImageModel(model, configuration)
+{
+  init_from_config(configuration, model->get_rt_info<ov::AnyMap>("model_info"));
+}
+
+AnomalyModel::AnomalyModel(std::shared_ptr<InferenceAdapter> &adapter)
+    : ImageModel(adapter)
+{
+  init_from_config(adapter->getModelConfig(), ov::AnyMap{});
+}
+
+std::unique_ptr<AnomalyResult> AnomalyModel::infer(
+    const ImageInputData &inputData)
+{
+  auto result = ModelBase::infer(static_cast<const InputData &>(inputData));
+
+  return std::unique_ptr<AnomalyResult>(
+      static_cast<AnomalyResult *>(result.release()));
+}
+
+std::unique_ptr<ResultBase> AnomalyModel::postprocess(
+    InferenceResult &infResult)
+{
+  ov::Tensor predictions = infResult.outputsData[outputNames[0]];
+  const auto &inputImgSize =
+      infResult.internalModelData->asRef<InternalImageModelData>();
+
+  double pred_score;
+  std::string pred_label;
+  cv::Mat anomaly_map;
+  cv::Mat pred_mask;
+  std::vector<cv::Rect> pred_boxes;
+  if (predictions.get_shape().size() == 1)
+  {
+    pred_score = predictions.data<float>()[0];
+  }
+  else
+  {
+    anomaly_map = cv::Mat(imageShape[0], imageShape[1], CV_32FC1,
+                          predictions.data<float>());
+    // find the max predicted score
+    cv::minMaxLoc(anomaly_map, NULL, &pred_score);
+  }
+  pred_label = labels[pred_score > imageThreshold ? 1 : 0];
+
+  if (task == "segmentation" || task == "detection")
+  {
+    pred_mask = anomaly_map >= pixelThreshold;
+    pred_mask.convertTo(pred_mask, CV_8UC1, 1/255.);
+    anomaly_map = normalize(anomaly_map, pixelThreshold);
+  }
+  pred_score = normalize(pred_score, imageThreshold);
+  if (!anomaly_map.empty())
+  {
+    cv::resize(
+        anomaly_map, anomaly_map,
+        cv::Size{inputImgSize.inputImgWidth, inputImgSize.inputImgHeight});
+    cv::resize(
+        pred_mask, pred_mask,
+        cv::Size{inputImgSize.inputImgWidth, inputImgSize.inputImgHeight});
+  }
+  if (task == "detection")
+  {
+    pred_boxes = getBoxes(pred_mask);
+  }
+
+  AnomalyResult *result =
+      new AnomalyResult(infResult.frameId, infResult.metaData);
+  result->anomaly_map = anomaly_map;
+  result->pred_score = pred_score;
+  result->pred_label = pred_label;
+  result->pred_mask = pred_mask;
+  result->pred_boxes = pred_boxes;
```

Comment:
You should either `std::move()` these values (except for `pred_score`) into `result->` or just use `result->*` directly without construction of intermediate vars throughput this func

Reply:
As I stated before, `pred_score` is an exception and shouldn't be `move`ed.

Reply:
No, strings are movable. The criteria is existenc of move constructor, which is (8) in https://en.cppreference.com/w/cpp/string/basic_string/basic_string

Reply:
Alright thanks! I think I'll go study C++ :)